interface/java_binding/src/test/java/org/khronos/ktx/test/KtxParallelTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023, Shukant Pal, robnugent, and Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.khronos.ktx.test;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.khronos.ktx.*;
+import java.util.Random;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ExtendWith({ KtxTestLibraryLoader.class })
+public class KtxParallelTest {
+    private static final int NUM_THREADS = 2;
+    private static final Logger logger = Logger.getLogger(KtxParallelTest.class.getCanonicalName());
+
+    @Test
+    public void testParallelAstcConversion() throws InterruptedException {
+        final Thread[] runThreads = new Thread[NUM_THREADS];
+
+        for (int i = 0; i < NUM_THREADS; i++) {
+            final KtxTestRun run = new KtxTestRun(i);
+            final Thread runThread = new Thread(run);
+            runThread.setDaemon(false);
+            runThread.start();
+
+            runThreads[i] = runThread;
+        }
+
+        for (Thread thread : runThreads) {
+            thread.join();
+        }
+    }
+
+    private static class KtxTestRun implements Runnable {
+        private final int id;
+        private final Random testRandomizer = new Random();
+
+        public KtxTestRun(int id) {
+            this.id = id;
+        }
+
+        public void run() {
+            // Repeatedly create a compress an image.
+            for (int i = 0; i < 1000; i++) {
```

Comment:
Reducing this to ~300 because the tests are timing out in CI. The crashes I saw locally happened after ~250 textures (when around ~1GB of heap memory was accumulated, triggering a GC I think)

```suggestion
            for (int i = 0; i < 300; i++) {
```